src/economy/portdock.cc

```diff
@@ -110,15 +110,12 @@ Flag& PortDock::base_flag() {
  * has the given flag.
  */
 PortDock* PortDock::get_dock(Flag& flag) const {
-	if (fleet_)
+	if (fleet_) {
 		return fleet_->get_dock(flag);
+	}
 	return nullptr;
 }
 
-uint32_t PortDock::get_need_ship() const {
```

Comment:
what happend to this part of the logic?

Reply:
Previously the responsibilities for naval decision-making were distributed between PortDock, Ship and ShipFleet. Now everything is handled by ShippingSchedule and the corresponding methods in other classes were deleted.